code/game/objects/items/ego_weapons/non_abnormality/rcorp.dm

```diff
@@ -34,3 +34,223 @@
+			L.apply_damage(force, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+
+/obj/item/ego_weapon/city/reindeer/captain
+	name = "R-corp reindeer captain staff"
+	icon_state = "rcorp_captainstaff"
+	force = 60
+	attribute_requirements = list(
+							FORTITUDE_ATTRIBUTE = 80,
+							PRUDENCE_ATTRIBUTE = 80,
+							TEMPERANCE_ATTRIBUTE = 80,
+							JUSTICE_ATTRIBUTE = 80
+							)
+
+
+//Shitty Raven Dagger
+/obj/item/ego_weapon/city/rabbit_blade/raven
+	name = "R-corp raven dagger"
+	attribute_requirements = list(
+							FORTITUDE_ATTRIBUTE = 40,
+							PRUDENCE_ATTRIBUTE = 40,
+							TEMPERANCE_ATTRIBUTE = 40,
+							JUSTICE_ATTRIBUTE = 40
+							)
+
+
+//Rcorp Guns
+
+/obj/item/gun/energy/e_gun/rabbit
+	name = "R-Corporation Lawnmower 3000"
+	desc = "An energy gun produced specifically to suppress threats within Lobotomy Corporation, it has four firing modes to switch between."
+	icon = 'ModularTegustation/Teguicons/lc13_weapons.dmi'
+	icon_state = "rabbit"
+	cell_type = /obj/item/stock_parts/cell/infinite
+	ammo_type = list(
+		/obj/item/ammo_casing/energy/laser/red,
+		/obj/item/ammo_casing/energy/laser/white,
+		/obj/item/ammo_casing/energy/laser/black,
+		/obj/item/ammo_casing/energy/laser/pale
+		)
+	can_charge = FALSE
+	weapon_weight = WEAPON_HEAVY // No dual wielding
+	pin = /obj/item/firing_pin/implant/mindshield
+
+/obj/item/gun/energy/e_gun/rabbit/Initialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.16 SECONDS)
+
+
+
+/obj/item/gun/energy/e_gun/rabbit/captain
+	name = "R-Corporation Lawnmower 4000"
+	desc = "An energy gun produced especially for the rabbit captain. This weapon can be fired with one hand."
+	icon_state = "rabbitcaptain"
+	weapon_weight = WEAPON_LIGHT
+	pin = /obj/item/firing_pin
+
+//you really shouldn't be having this as a spawned in rabbit
+/obj/item/gun/energy/e_gun/rabbit/nopin
+	name = "R-Corporation Lawnmower 2700"
+	desc = "An energy gun produced specifically to suppress threats with a variety of damage types. This one is an older model, and only has 3 modes."
+	ammo_type = list(
+		/obj/item/ammo_casing/energy/laser/red,
+		/obj/item/ammo_casing/energy/laser/white,
+		/obj/item/ammo_casing/energy/laser/black
+		)
+	pin = /obj/item/firing_pin
+
+
+/obj/item/gun/energy/e_gun/rabbitdash
+	name = "R-Corporation Lawnmower 2000"
+	desc = "An energy gun mass-produced by R corporation for the bulk of their force."
+	icon = 'ModularTegustation/Teguicons/lc13_weapons.dmi'
+	icon_state = "rabbitk"
+	fire_delay = 5
+	cell_type = /obj/item/stock_parts/cell/infinite
+	ammo_type = list(
+		/obj/item/ammo_casing/energy/laser/red,
+		)
+	can_charge = FALSE
+	weapon_weight = WEAPON_HEAVY // No dual wielding
+	pin = /obj/item/firing_pin
+
+/obj/item/ego_weapon/city/rabbit_rush
+	name = "rush dagger"
+	desc = "A high-frequency combat blade made for use against abnormalities and other threats in Lobotomy Corporation and the outskirts. This only has one mode"
```

Comment:
This one does not have it's abilities listed anywhere. Perhaps add it in `special` as it inherits from `ego_weapon`?

Reply:
Right, I will add that